### PR TITLE
improved implementation - all test passing now

### PR DESCRIPTION
**Pull request by kituuu**
### Describe your change:

* [x] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by cclauss** [comment]
A machine does `pre-commit autoupdate` for us... #6940

File: matrix/inverse_of_matrix.py [hunk 1]
```diff
@@ -25,24 +26,107 @@ def inverse_of_matrix(matrix: list[list[float]]) -> list[list[float]]:
...
     >>> inverse_of_matrix([[10, 5], [3, 2.5]])
     [[0.25, -0.5], [-0.3, 1.0]]
+
+    Demo for 3x3
+    >>> inverse_of_matrix([[2, 5, 7], [2, 0, 1], [1, 2, 3]])
+    [[2.0, 1.0, -5.0], [5.0, 1.0, -12.0], [-4.0, -1.0, 10.0]]
+    >>> inverse_of_matrix([[1, 2, 2], [1, 2, 2], [3, 2, -1]])
+    Traceback (most recent call last):
+    ...
+    ValueError: This matrix has no inverse.
```
Comment (cclauss) [inline matrix/inverse_of_matrix.py:36]: Please add four more doctests:
```
>>> inverse_of_matrix([])
>>> inverse_of_matrix([[], []])
>>> inverse_of_matrix([[1, 2], [3, 4], [5, 6]])
>>> inverse_of_matrix([[1, 2, 3], [4, 5, 6]])
```

Comment (kituuu) [reply matrix/inverse_of_matrix.py:36]: Ok sir, I am adding

**Comment by dhruvmanila** [comment]
@algorithms-keeper review-all

If possible, please fix them.

**Comment by kituuu** [comment]
Sir, I made the changes and improved the docstring, but I don't know why installing dependencies is failing 😐. Please review it and help If done something wrong

File: matrix/inverse_of_matrix.py (hunk 1 above)
Comment (kituuu) [reply matrix/inverse_of_matrix.py:36]: Done sir

**Comment by kituuu** [comment]
If everything is fine please give it a hacktoberfest label :D Thank you

**Comment by kituuu** [comment]
Hey anyone here ??
Please review it :((

**Comment by kituuu** [comment]
@cclauss @dhruvmanila Sir can u PLEASE review this PR of mine?

**Pull request closed without merging**